packages/cli/src/utils/config.ts

```diff
@@ -1,24 +1,149 @@
-// TODO unfinished placeholder
+import { findUpSync } from "find-up";
+import path from "path";
+import { SchemaType } from "@latticexyz/schema-type";
+
+// TODO require may or may not be needed, finish exploring the various configurations
+import { createRequire } from "module";
+import { fileURLToPath } from "url";
+import { ERRORS, MUDError } from "./errors";
+import { defaultKeyTuple, defaultStoreImportPath, defaultTablePath } from "./constants";
+const require = createRequire(fileURLToPath(import.meta.url));
+
+// Based on hardhat's config (MIT)
+// https://github.com/NomicFoundation/hardhat/tree/main/packages/hardhat-core
+
+const TS_CONFIG_FILENAME = "mud.config.mts";
+
 export interface StoreUserConfig {
-  storePath?: string;
+  /** Path for store package imports. Default is "@latticexyz/store/src/" */
+  storeImportPath?: string;
+  /** Configuration for each table. The keys are table names, 1st letter should be uppercase. */
   tables: {
-    [name: string]: {
-      keys?: string[];
+    [tableName: string]: {
+      /** Output path for the file. Default is "tables/" */
+      path?: string;
+      /** List of names for the table's keys. Default is ["key"] */
+      keyTuple?: string[];
+      /** Table's columns. The keys are column names, 1st letter should be lowercase. */
       schema: {
-        [valueName: string]: number;
+        [columnName: string]: SchemaType;
       };
     };
   };
 }
 
 export interface StoreConfig extends StoreUserConfig {
-  storePath: string;
+  storeImportPath: string;
   tables: {
-    [name: string]: {
-      keys: string[];
+    [tableName: string]: {
+      path: string;
+      keyTuple: string[];
       schema: {
-        [valueName: string]: number;
+        [columnName: string]: SchemaType;
       };
     };
   };
 }
+
+export async function loadConfig({ configPath }: { configPath?: string }) {
+  configPath = resolveConfigPath(configPath);
+
+  // TODO is this override really necessary?
+  process.env.TS_NODE_COMPILER_OPTIONS = `{ "module": "esnext" }`;
```

Comment:
what's the intention for this override?

Reply:
(this is now in `storeConfig.ts`)

Reply:
I think it makes schema-type import in user config work well regardless of user's tsconfig. TODO is about me thoroughly figuring out the exact situations it's needed and possible workarounds